Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride.hs

```diff
@@ -198,53 +209,67 @@ mkDriverRideRes ::
   Maybe DSM.BapMetadata ->
   Maybe (Id DDGR.DriverGoHomeRequest) ->
   Maybe DI.DriverInformation ->
-  DriverRideRes
+  m DriverRideRes
 mkDriverRideRes rideDetails driverNumber rideRating mbExophone (ride, booking) bapMetadata goHomeReqId driverInfo = do
   let fareParams = booking.fareParams
       estimatedBaseFare =
         fareSum $
           fareParams{driverSelectedFare = Nothing -- it should not be part of estimatedBaseFare
                     }
   let initial = "" :: Text
-  DriverRideRes
-    { id = ride.id,
-      shortRideId = ride.shortId,
-      status = ride.status,
-      fromLocation = DLoc.makeLocationAPIEntity booking.fromLocation,
-      toLocation = DLoc.makeLocationAPIEntity <$> booking.toLocation,
-      driverName = rideDetails.driverName,
-      driverNumber,
-      vehicleNumber = rideDetails.vehicleNumber,
-      vehicleColor = fromMaybe initial rideDetails.vehicleColor,
-      vehicleVariant = fromMaybe DVeh.SEDAN rideDetails.vehicleVariant,
-      vehicleModel = fromMaybe initial rideDetails.vehicleModel,
-      computedFare = ride.fare,
-      estimatedBaseFare = estimatedBaseFare,
-      estimatedDistance = booking.estimatedDistance,
-      driverSelectedFare = fromMaybe 0 fareParams.driverSelectedFare,
-      actualRideDistance = ride.traveledDistance,
-      createdAt = ride.createdAt,
-      updatedAt = ride.updatedAt,
-      riderName = booking.riderName,
-      pickupDropOutsideOfThreshold = ride.pickupDropOutsideOfThreshold,
-      tripStartTime = ride.tripStartTime,
-      tripEndTime = ride.tripEndTime,
-      specialLocationTag = booking.specialLocationTag,
-      rideRating = rideRating <&> (.ratingValue),
-      chargeableDistance = ride.chargeableDistance,
-      exoPhone = maybe booking.primaryExophone (\exophone -> if not exophone.isPrimaryDown then exophone.primaryPhone else exophone.backupPhone) mbExophone,
-      customerExtraFee = fareParams.customerExtraFee,
-      bapName = bapMetadata <&> (.name),
-      bapLogo = bapMetadata <&> (.logoUrl),
-      disabilityTag = booking.disabilityTag,
-      requestedVehicleVariant = booking.vehicleVariant,
-      driverGoHomeRequestId = goHomeReqId,
-      payerVpa = driverInfo >>= (.payerVpa),
-      autoPayStatus = driverInfo >>= (.autoPayStatus),
-      isFreeRide = ride.isFreeRide,
-      customerCancellationDues = fareParams.customerCancellationDues,
-      stopLocationId = (.id) <$> booking.toLocation
-    }
+  maxOrder <- QLM.maxOrderByEntity booking.id.getId
```

Comment:
We should not be doing this in case of normal bookings. Lets only do it booking.tripCategory is of rental type,